src/python/pants/scm/subsystems/changed.py

```diff
@@ -42,7 +52,41 @@ def register_options(cls, register):
              help='Calculate changes since this tree-ish/scm ref (defaults to current HEAD/tip).')
     register('--diffspec',
              help='Calculate changes contained within given scm spec (commit range/sha/ref/etc).')
-    register('--include-dependees', choices=['none', 'direct', 'transitive'], default='none',
+    register('--include-dependees', type=IncludeDependees, default=IncludeDependees.none,
              help='Include direct or transitive dependees of changed targets.')
     register('--fast', type=bool,
              help='Stop searching for owners once a source is mapped to at least one owning target.')
+
+
+@dataclass(frozen=True)
+class ScmWrapper:
+  scm: Scm
+
+
+@dataclass(frozen=True)
+class ChangedFilesResult:
+  changed_files: Tuple[str, ...]
+
+
+# TODO: ensure this rule isn't ever cached with an @ensure_daemon integration test!
```

Comment:
Hm, shoot. Forgot about this. This is a blocker for landing.

The issue with using a hack based on injecting the SCM as a `RootRule` with an `__eq__` method that will never match (previously discussed [here](https://github.com/pantsbuild/pants/pull/7350#discussion_r272434402)) is that that would cause us to re-run all of the `@rules` transitively from the root down to use-sites of the SCM... even if a particular request didn't _actually_ consume the SCM.

An alternative might be to parse enough of the `--owners-of`, `--changed`, and the specs to know which were non-empty, and then inject the non-empty thing as the RootRule?

Reply:
The other alternative is to lean in to implement #6598, and then mark the @rule that touches the SCM as uncacheable.